engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java

```diff
@@ -192,6 +201,22 @@ public QualifiedName getName() {
     return name;
   }
 
+  /**
+   * Renames a project part of the QualifiedName of this module.
+   *
+   * @param oldName the old project name
+   * @param newName the new project name
+   * @return a module with the updated QualifiedName
+   */
+  public Module renameProject(String oldName, String newName) {
```

Comment:
Why not make this change mutably? This way we could avoid losing the cache and make sure all the precomputed values update the names accordingly. The full object swap is dangerous, as there exist lower-level objects holding on to this reference and may not get updated properly (using a phantom module instead).

Reply:
Since we decided not to change it mutably can you approve the PR?